### ISSUE TITLE
Skip persisting empty commits (with 0 snapshots) in SQL repository 

### ISSUE THREAD

**Issue by mdii** [javers/javers#505]
Hi,

I am using JaVers 3.0.0 with Spring Boot. I noticed that new entries are stored in jv_commit even when there is no change in the object. Since these commits are not referenced from other tables, they are kind of useless.

**Comment by bartoszwalacik** [comment]
Hi, please follow the https://github.com/javers/javers/blob/master/CONTRIBUTING.md

Bug Reports have to contain:

    - Clear description of your expectations versus reality
    - Runnable test case which isolates the bug and allows us to easily reproduce it on our laptops. You can push this test case to your fork of this repository.

**Comment by mdii** [comment]
As you can see in the following example, the second commit creates an entry in jv_commit and no entries in jv_snapshots. When we auto-audit repositories this leads to a lot of entries in jv_commit that are not referenced from other tables because usually we don't check if the entity has been updated before calling repository.save(entity).

```java
Javers javers = JaversBuilder.javers().build();
Person robert = new Person("bob", "Robert Martin");
javers.commit("user", robert); 

//creates an entry in jv_commit even though nothing has changed on the object
javers.commit("user", robert); 
```

The expected behavior would be that commits are persisted only when the commit has snapshots.

Sorry, I couldn't come up with a runnable test case.

**Comment by bartoszwalacik** [comment]
You mean this is not OK that *empty* commits are persisted?

**Comment by mdii** [comment]
Yes, exactly.

**Comment by bartoszwalacik** [comment]
Feel free to create a pull request. This issue is easy to implement so perfect for contribution.

**Comment by mdii** [comment]
I can create a pull request. 
Should this be applied only to SQL repository or to all repositories?

**Comment by bartoszwalacik** [comment]
PR should be created from your fork. Looks like the problem could be solved in javerscore, before calling JaversRepository

**Comment by bartoszwalacik** [comment]
PR merged to master

**Comment by bartoszwalacik** [comment]
@mdii your commits are not bounded to your user. If you fix it, you will be listed on javers.org as a contributor

https://github.com/javers/javers/commit/d81eb3c63644c89a1c64b471744b9b60d94dc840

**Comment by mdii** [comment]
It's done, thanks!

**Comment by bartoszwalacik** [comment]
released in 3.0.4
